zipserver/archive.go

```diff
@@ -133,7 +134,7 @@ type UploadFileResult struct {
 	Size  uint64
 }
 
-func uploadWorker(a *Archiver, limits *ExtractLimits, tasks <-chan UploadFileTask, results chan<- UploadFileResult, cancel chan struct{}, done chan struct{}) {
+func uploadWorker(a *Archiver, limits *ExtractLimits, tasks <-chan UploadFileTask, results chan<- UploadFileResult, done chan struct{}) {
```

Comment:
> you can pass in the new context object down to the http PUT request

That would be appropriate, so I'll do that. In general it's good to propagate cancellations to sub-requests etc.

Now that we're talking about it, these Contexts should actually be children of the incoming HTTP request Context, in case the client disconnects. (We don't want that in the async processing case though.)